IntegrationWithPharmacies/Controllers/SharingPrescriptionController.cs

```diff
@@ -1,30 +1,199 @@
+        {
+            return Ok(PatientService.GetAll());
+        }
+        [HttpGet]
+        public IActionResult Get()
+        {
+            return Ok(MedicineService.GetAll());
+        }
+
+        [HttpGet("medicinesIsa")]
+        public IActionResult GetMedicinesFromIsa()
+        {
+            var client = new RestSharp.RestClient("http://localhost:8082");
+            var request = new RestRequest("/medicineRequested");
+            var response = client.Get<List<MedicineName>>(request);
+            Console.WriteLine("Status: " + response.StatusCode.ToString());
+            List<MedicineName> result = response.Data;
+            result.ForEach(medicine => Console.WriteLine(medicine.ToString()));
+            return Ok(result);
+        }
+
+        [HttpPost]
+        public IActionResult Post(Prescription prescription)
+        {
+            if (Environment.Equals("Local")) return sendPrescriptionSftp(prescription);
+            return BadRequest();
+        }
+
+        private IActionResult sendPrescriptionSftp(Prescription prescription)
+        {
+            var sftpService = new SftpService(new NullLogger<SftpService>(), getConfig());
+            String complete = @"FilePrescriptions\..\Prescription" + DateTime.Now.ToString("dd-MM-yyyy") + "_" + getRandomNumber() + ".txt";
+            System.IO.FileStream fs = System.IO.File.Create(complete);
+            fs.Close();
+            System.IO.File.WriteAllText(complete, getTextForPrescription(prescription));
+            sftpService.UploadFile(@"..\test.txt", @"\pub\" + complete);
+            SendNotificationAboutReport();
+            return Ok();
+        }
+
+        [HttpGet("http/recieve/{medicine}")]
+        public IActionResult GetMedicineDescription(string medicine)
+        {
+            var url = "http://localhost:8082/upload/medicine/" + medicine;
+            HttpWebRequest webRequest = (HttpWebRequest)WebRequest.Create(url);
+            HttpWebResponse webResponse = (HttpWebResponse)webRequest.GetResponse();
+            Stream response = webResponse.GetResponseStream();
+            StreamReader readStream = new StreamReader(response, System.Text.Encoding.GetEncoding("utf-8"));
+            string description = readStream.ReadToEnd();
+            if (description.Length != 0) return Ok(description);
+            return BadRequest();
+            
+        }
+        [HttpGet("http/medicineAvailability/{medicine}")]
+        public IActionResult GetMedicineAvailability(String medicine)
+        {
+            HttpWebRequest webRequest = (HttpWebRequest)WebRequest.Create("http://localhost:8082/medicinePharmacy/" + medicine);
+            HttpWebResponse webResponse = (HttpWebResponse)webRequest.GetResponse();
+            Stream response = webResponse.GetResponseStream();
+            StreamReader readStream = new StreamReader(response, System.Text.Encoding.GetEncoding("utf-8"));
+            string availability = readStream.ReadToEnd();

```

Comment:
Change return statement to return FormMedicineAvailability(readStream.ReadToEnd())